astropy/config/configuration.py

```diff
@@ -43,14 +40,6 @@ class InvalidConfigurationItemWarning(AstropyWarning):
     """
 
 
-class ConfigurationMissingWarning(AstropyWarning):
```

Comment:
Removing this might break a tiny handful of existing scripts that deliberately import `ConfigurationMissingWarning` in order to suppress it.  I'm not sure if there is a great (existing) way we could create a deprecation warning for it in this case though.

Reply:
Maybe this could just be noted in the changelog.

Reply:
Right, we can keep it for now. Btw I should probably add the deprecation warning for `update_default_config` ?